File: AN_2/SEM_1/MAP/lab2/src/model/SortingTask.java
package model;
import Enum.SortStrategy;
import java.util.Arrays;

public class SortingTask extends Task
{
    private AbstractSorter abstractSorter;
    private int[] numere;
    private SortStrategy sortStrategy;

    public SortingTask(String id_task, String descriere, int[] numere, SortStrategy sortStrategy)
    {
        super(id_task, descriere);
        this.numere = numere;
        this.sortStrategy = sortStrategy;

        switch (sortStrategy)
        {
            case BUBBLESORT -> abstractSorter = new BubbleSort();
            case QUICKSORT -> abstractSorter = new QuickSort();
            default -> {}
        }
    }

    public void execute()
    {
        abstractSorter.sort(numere);
        System.out.println(Arrays.toString(numere));
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Model/SortingTask.cs
using System;
using System.Linq;
using Enum;

namespace Model
{
    public class SortingTask : Tasks
    {
        private AbstractSorter abstractSorter;
        private int[] numere;
        private SortStrategy sortStrategy;

        public SortingTask(string id_task, string description, int[] numere, SortStrategy sortStrategy)
            : base(id_task, description)
        {
            this.numere = numere;
            this.sortStrategy = sortStrategy;

            switch (sortStrategy)
            {
                case SortStrategy.BubbleSort:
                    abstractSorter = new BubbleSort();
                    break;
                case SortStrategy.QuickSort:
                    abstractSorter = new QuickSort();
                    break;
                default:
                    break;
            }
        }

        public override void Execute()
        {
            abstractSorter.Sort(numere);
            Console.WriteLine(string.Join(", ", numere));
        }
    }
}
